### ISSUE TITLE
Uncaught TypeError: Cannot read property 'triggerHandler' of null

### ISSUE THREAD

**Issue by xunhuanhuan** [ezraroi/ngJsTree#84]

**Comment by ezraroi** [comment]
Please provide code that reproduce

**Comment by xunhuanhuan** [comment]
app.controller("treeCtrl", function ($http,$scope,$log,$timeout) {
    var vm = this;
    var newId = 1;
    vm.ignoreChanges = false;
    vm.newNode = {};
    $http({
        url:'json/1.js',
        method:'GET',
        dataType: 'json'
        }).success(function(data,header,config,status){
           vm.originalData = data;
           callback();
        }).error(function(data,header,config,status){
    });
    vm.originalData = [];
    var callback = function() {
        vm.treeData = [];
        angular.copy(vm.originalData,vm.treeData);
        vm.treeConfig = {
            core : {
                multiple : false,
                animation: true,
                error : function(error) {
                    $log.error('treeCtrl: error from js tree - ' + angular.toJson(error));
                },
                check_callback : true,
                worker : true
            },
            version : 1,
            plugins : ['types','checkbox']
        };
        vm.addNewNode = function() {
            vm.treeData.push({ id : (newId++).toString(), parent : vm.newNode.parent, text : vm.newNode.text });
        };
        this.applyModelChanges = function() {
                return !vm.ignoreChanges;
        };
    }
})

**Comment by xunhuanhuan** [comment]
![c9f0 tmp](https://cloud.githubusercontent.com/assets/15319838/16827918/8862e4f0-49bd-11e6-9bfe-27c8536c046f.jpg)
![e0f6 tmp](https://cloud.githubusercontent.com/assets/15319838/16827917/8862ca88-49bd-11e6-831f-0eebcdb0c082.jpg)
![22e0 tmp](https://cloud.githubusercontent.com/assets/15319838/16827919/88695c54-49bd-11e6-85a4-717aa59ea77c.jpg)

**Comment by xunhuanhuan** [comment]
Click the button can not be increased

**Comment by ezraroi** [comment]
Please reproduce in plunker so i will be able to check

**Comment by DoubleDot** [comment]
I had the same issue, async loading the items from the backend. When removing the "treeConfig.version++" it works, but it doesn't update the view, as 'expected'.

I changed my html to do the following, to ensure everything is loaded before loading the directive.

`                        <div ng-if="treeData != undefined">
                            <div js-tree="treeConfig" ng-model="treeData" tree="treeInstance" tree-events="changed:changedCB"></div>
                        </div>`

I tried all the fixes provided here. Nothing seems to work except for the method above.
